graylog2-server/src/main/java/org/graylog2/shared/rest/resources/ProxiedResource.java

```diff
@@ -32,4 +53,45 @@ public ProxiedResource(@Context HttpHeaders httpHeaders) {
             this.authenticationToken = null;
         }
     }
+
+    protected <T, K> Map<String, Optional<K>> getForAllNodes(Function<T, Call<K>> fn, Function<String, Optional<T>> interfaceProvider) {
+        return getForAllNodes(fn, interfaceProvider, Function.identity());
+    }
+
+    protected <T, K, L> Map<String, Optional<K>> getForAllNodes(Function<T, Call<L>> fn, Function<String, Optional<T>> interfaceProvider, Function<L, K> transformer) {
+        return this.nodeService.allActive()
+            .entrySet()
+            .stream()
```

Comment:
Hm, somehow the comment I had already written was either not saved or got lost.

I left it out for now, because it is inconsistently used and I am not 100% sure that we can exclude negative implications. The benefit of this change though is that we have a central point to switch it on/off if we want to.